frontend/src/components/Connection/ConnectionSelector.tsx

```diff
@@ -2,16 +2,15 @@ import ConnectionImage from "./DatabaseDialectImage";
 import { useState } from "react";
 import { IConnection, IConversation } from "../Library/types";
 import { Cog6ToothIcon } from "@heroicons/react/24/outline";
-import { Link, useNavigate } from "react-router-dom";
-import { Routes } from "@/router";
 import { useCreateConversation, useGetConnections } from "@/hooks";
+import { Link, useNavigate } from "@tanstack/react-router";
 
 export const ConnectionSelector = () => {
   const navigate = useNavigate();
   const [, setConversation] = useState<IConversation | null>();
   const { data } = useGetConnections();
   const createConnection = () => {
-    navigate(Routes.NewConnection);
+    navigate({ to: "/connection/new" });
```

Comment:
Why did you remove the constants here? Didn't get it.

Reply:
1- URLs in the files created inside `src/routes` are auto-generated/auto-modified based on the relative paths of the files and their file names. For example, if I simply create a file called `hello.tsx` inside `src/routes/chat/hello`, the URL will automatically be `"/chat/hello"`, and if I try to modify the URL, we get an error

2- URL strings are now autocompleted

![image](https://github.com/RamiAwar/dataline/assets/26882839/13f2bb4e-01a4-4b9b-bfae-2843abd91ab9)

Reply:
Okay, I really don't like this tho. Intellisense should help and not be crucial to coding. Is there a non-file based routing option? I don't see the point of having it be file based if all the files are empty and referencing a single component. It's so much fluff and downsides with no upsides for DataLine so far.

Reply:
What downsides?

Reply:
Like not having clarity as to how routing is working cause it's so distributed across many files making it harder to find stuff, and not having constants for the strings and potentially introducing bugs if you don't have intellisense